internal/operator-controller/features/features.go

```diff
@@ -36,3 +39,20 @@ var OperatorControllerFeatureGate featuregate.MutableFeatureGate = featuregate.N
 func init() {
 	utilruntime.Must(OperatorControllerFeatureGate.Add(operatorControllerFeatureGates))
 }
+
+// LogFeatureGateStates logs the state of all known feature gates.
+func LogFeatureGateStates(log logr.Logger, fg featuregate.FeatureGate) {
```

Comment:
We also have catalogd feature gates that would need to be logged. We need consistency between them to keep our sanity.

Reply:
Are you wanting us to add that to this PR? Sounds like it could be a contentious addition

Reply:
Let's capture as a follow-up. In an ideal world the whole feature gate logging change would have been a separate PR where we add logging to both catalogd and operator-controller at the same time as its own change.

Reply:
Follow-up issue https://github.com/operator-framework/operator-controller/issues/1922